### PR TITLE
[aznamespaces] Moving to new folder site, updating readme and autorest

### PR DESCRIPTION
**Pull request by richardpark-msft**
Moving the package for Event Grid namespaces into it's own module folder, sibling to the other Event Grid packages that are coming (azsystemevents and azeventgrid, which will contain the client for "Event Grid Basic").

### CODE REVIEW

**Comment by richardpark-msft** [comment]
@jhendrixMSFT  - the failures for the older azeventgrid pipeline can be ignored. Just an artifact of me moving these things around behind the back of the CI pipelines.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
I'd like to understand the release logistics and migration strategy.

Do we need to release an updated `azeventgrid` with a deprecation notice that users need to migrate to `aznamespaces`?
What's our obligation to fix issues in the existing `azeventgrid`? i.e. is there a window of time where we're required to, or can we just say "move to `aznamespaces`"?
Will we reset the version number to `v0.1.0` for `aznamespaces`?

**Comment by richardpark-msft** [comment]
> I'd like to understand the release logistics and migration strategy.
> 
> Do we need to release an updated `azeventgrid` with a deprecation notice that users need to migrate to `aznamespaces`? 

Yes, the old package will just get one more publish to mark it as deprecated.

> What's our obligation to fix issues in the existing `azeventgrid`? i.e. is there a window of time where we're required to, or can we just say "move to `aznamespaces`"? 

It was a beta so I don't think we have any obligation - I'll just point them to aznamespaces.

> Will we reset the version number to `v0.1.0` for `aznamespaces`?

I wasn't thinking we'd reset it, but we could. Either way it'll have to go for one more beta since I need to add in some missing functionality.

**Comment by richardpark-msft** [comment]
@jhendrixMSFT, I brought back the old modules files.

File: sdk/messaging/eventgrid/aznamespaces/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Release History
+
+## 0.4.1 (Unreleased)
+
+### Features Added
+
+### Breaking Changes
+
+- This module has been moved from it's previous location in `azeventgrid` to this location (`github.com/Azure/azure-sdk-for-go/sdk/messaging/eventgrid/aznamespaces`).
```
Comment (chlowell) [inline sdk/messaging/eventgrid/aznamespaces/CHANGELOG.md:9]:
```suggestion
- This module has been moved from its previous location in `azeventgrid` to this location (`github.com/Azure/azure-sdk-for-go/sdk/messaging/eventgrid/aznamespaces`).
```

File: sdk/messaging/eventgrid/aznamespaces/README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+[cloud_shell_bash]: https://shell.azure.com/bash
+[source]: https://aka.ms/azsdk/go/namespaces/src
+[godoc]: https://aka.ms/azsdk/go/namespaces/pkg
+[godoc_client]: https://aka.ms/azsdk/go/namespaces/pkg/#Client
+[godoc_client_publish]: https://aka.ms/azsdk/go/namespaces/pkg#Client.PublishCloudEvents
+[godoc_client_receive]: https://aka.ms/azsdk/go/namespaces/pkg#Client.ReceiveCloudEvents
+[godoc_examples]: https://aka.ms/azsdk/go/namespaces/pkg#pkg-examples
+[godoc_example_newclient]: https://aka.ms/azsdk/go/namespaces/pkg#example-NewClientWithSharedKeyCredential
+[godoc_publisher_client]: https://aka.ms/azsdk/go/eventgrid/pkg/#Client
+[ms_pulldelivery]: https://learn.microsoft.com/azure/event-grid/concepts-pull-delivery
```
Comment (chlowell) [inline sdk/messaging/eventgrid/aznamespaces/README.md:123]:
```suggestion
```

File: sdk/messaging/eventgrid/aznamespaces/README.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+[azure_cli]: https://docs.microsoft.com/cli/azure
+[azure_pattern_circuit_breaker]: https://docs.microsoft.com/azure/architecture/patterns/circuit-breaker
+[azure_pattern_retry]: https://docs.microsoft.com/azure/architecture/patterns/retry
+[azure_portal]: https://portal.azure.com
+[azure_sub]: https://azure.microsoft.com/free/
+[cloud_shell]: https://docs.microsoft.com/azure/cloud-shell/overview
+[cloud_shell_bash]: https://shell.azure.com/bash
+[source]: https://aka.ms/azsdk/go/namespaces/src
+[godoc]: https://aka.ms/azsdk/go/namespaces/pkg
+[godoc_client]: https://aka.ms/azsdk/go/namespaces/pkg/#Client
```
Comment (chlowell) [inline sdk/messaging/eventgrid/aznamespaces/README.md:117]:
```suggestion
```

File: sdk/messaging/eventgrid/aznamespaces/README.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+[azure_sdk_for_go_contributing]: https://github.com/Azure/azure-sdk-for-go/blob/main/CONTRIBUTING.md
+[azure_sdk_for_go_contributing_developer_guide]: https://github.com/Azure/azure-sdk-for-go/blob/main/CONTRIBUTING.md#developer-guide
+[azure_sdk_for_go_contributing_pull_requests]: https://github.com/Azure/azure-sdk-for-go/blob/main/CONTRIBUTING.md#pull-requests
+[azure_cli]: https://docs.microsoft.com/cli/azure
+[azure_pattern_circuit_breaker]: https://docs.microsoft.com/azure/architecture/patterns/circuit-breaker
+[azure_pattern_retry]: https://docs.microsoft.com/azure/architecture/patterns/retry
+[azure_portal]: https://portal.azure.com
+[azure_sub]: https://azure.microsoft.com/free/
+[cloud_shell]: https://docs.microsoft.com/azure/cloud-shell/overview
+[cloud_shell_bash]: https://shell.azure.com/bash
```
Comment (chlowell) [inline sdk/messaging/eventgrid/aznamespaces/README.md:114]:
```suggestion
```

File: sdk/messaging/eventgrid/aznamespaces/build.go [hunk 5]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+//go:build go1.18
+// +build go1.18
```
Comment (chlowell) [inline sdk/messaging/eventgrid/aznamespaces/build.go:4]:
```suggestion
```

**Comment by RickWinter** [comment]
> > I'd like to understand the release logistics and migration strategy.
> > Do we need to release an updated `azeventgrid` with a deprecation notice that users need to migrate to `aznamespaces`?
> 
> Yes, the old package will just get one more publish to mark it as deprecated.
> 
> > What's our obligation to fix issues in the existing `azeventgrid`? i.e. is there a window of time where we're required to, or can we just say "move to `aznamespaces`"?
> 
> It was a beta so I don't think we have any obligation - I'll just point them to aznamespaces.
> 
> > Will we reset the version number to `v0.1.0` for `aznamespaces`?
> 
> I wasn't thinking we'd reset it, but we could. Either way it'll have to go for one more beta since I need to add in some missing functionality.

It was a Beta so we can deprecate at our discretion.

I'd be in favor of starting at `v0.1.0`.  Gives a good impression of this is the new thing.

File: sdk/messaging/eventgrid/aznamespaces/go.mod [hunk 6]
```diff
@@ -0,0 +1,29 @@
+module github.com/Azure/azure-sdk-for-go/sdk/messaging/eventgrid/aznamespaces
+
+go 1.20
```
Comment (jhendrixMSFT) [inline sdk/messaging/eventgrid/aznamespaces/go.mod:3]: Should be `1.18`, is this intentional?

**Comment by jhendrixMSFT** [comment]
Is this identical to `azeventgrid`?

**Comment by richardpark-msft** [comment]
> Member

Yes, minus the subpackage `publisher`.

**Comment by jhendrixMSFT** [comment]
Does `azeventgrid/publisher` go away completely, or does that end up in `aznamespaces`?

**Comment by richardpark-msft** [comment]
> Does `azeventgrid/publisher` go away completely, or does that end up in `aznamespaces`?

Publisher is going to be it's own module. So the package map, as of now:

* `messaging/azeventgrid/publisher` -> `messaging/eventgrid/azeventgrid` (this matches what the other languages have used the main Event Grid package for)
* `messaging/azeventgrid` -> `messaging/eventgrid/aznamespaces` (this is the new service)

We also now have an messaging/eventgrid/azsystemevents module as well which I hadn't published in any form yet.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**